### DISCUSSION TITLE
Disable treesitter if lsp supports semantic tokens

### DISCUSSION THREAD

**Discussion by KillTheMule** [neovim/neovim#25802] [General]
So my syntax highlighting became more of a christmas-tree thing, and at least part of the blame lies certainly on the fact that both treesitter AND the lsp do highlighting. As far as I can see, I want to only have the lsp highlighting, but I'd like to solve this sort-of-properly.

I've seen I can disable treesitter bufferwise by using `vim.treesitter.stop()`, so I think I have to call this `on_attach`. I am, however, unclear about how to find out if the server actually supports semantic tokens, which would be a prerequisite for disabling treesitter.

So, can anyone tell me how to find out, preferably in `on_attach`? 

Moreover, is this a good way to achieve that? Or is there a better solution I'm not aware of? Thanks in advance!

**Comment by clason** [comment]
You need to check the (negotiated) capabilities `:h vim.lsp.get_active_clients()`.

**Reply by KillTheMule** [reply]
Thanks for the hint! If anyone's interested, it goes like this:

```lua
local on_attach = function(client, bufnr)
  if client.server_capabilities.semanticTokensProvider then
    vim.treesitter.stop(bufnr)
  end
end
```

where `on_attach` fits into `lspconfig`'s `setup` functionality.
